tests/network-ovn

```diff
@@ -2096,6 +2096,51 @@ ovn_leases_tests() {
         lxc network delete lxdbr0
 }
 
+ovn_vlan_uplink_tests() {
+    echo "==> vlan uplink"
+
+    sudo ovs-vsctl set open_vswitch . \
+    external_ids:ovn-remote=unix:/var/run/ovn/ovnsb_db.sock \
+    external_ids:ovn-encap-type=geneve \
+    external_ids:ovn-encap-ip=127.0.0.1
+
+    lxc network create lxdbr0 \
+        ipv4.address=10.10.10.1/24 ipv4.nat=true \
+        ipv4.dhcp.ranges=10.10.10.2-10.10.10.199 \
+        ipv4.ovn.ranges=10.10.10.200-10.10.10.254 \
+        ipv6.address=fd42:4242:4242:1010::1/64 ipv6.nat=true \
+        ipv6.ovn.ranges=fd42:4242:4242:1010::200-fd42:4242:4242:1010::254
+
+    echo "==> Create OVN network using a specified VLAN ID"
+    lxc network create ovn-virtual-network --type=ovn network=lxdbr0 vlan=10
+    sleep 2
+
+    echo "==> Ensure localnet port has the tag column set in OVN northbound DB"
+    ovn-nbctl --bare --format=csv --columns=type,tag find logical_switch_port | grep -F "localnet,10"
+
+    echo "==> Ensure OVS configured to accept VLAN traffic"
+    uplink_port=$(ovs-vsctl show | grep "Bridge lxdovn" | awk '{print $2"b"}')
+    ovs-vsctl get port "${uplink_port}" vlan_mode | grep -F "trunk"
+    ovs-vsctl get port "${uplink_port}" trunks | grep -F "[10]"
+
+    echo "==> Ensure OVN network can reach containers on the configured VLAN"
+    lxc launch "${IMAGE}" c1-ovn -n ovn-virtual-network
+    lxc launch "${IMAGE}" c1-lxdbr0 -n lxdbr0
+    sleep 2
+
+    lxc config device set c1-lxdbr0 eth0 vlan=10
+    lxc exec c1-lxdbr0 -- ip addr add 10.10.10.100/24 dev eth0
+    lxc exec c1-lxdbr0 -- ip link set dev eth0 up
```

Comment:
In my tests, the device remains up even after adding the `vlan=10` config so this should not be needed.

Reply:
If I don't ensure `eth0` is up then I see the next command to add the default route fail:
```
$ lxc exec c1-lxdbr0 -- ip route add default via 10.10.10.1
Error: Nexthop has invalid gateway.
```

Am I missing something here?

Reply:
Hmm, sounds like a race as the `up` part is not needed locally but I've found some other occurrences https://serverfault.com/questions/953198/why-nexthop-has-invalid-gateway-when-it-seems-to-be-defined

It doesn't hurt so please keep it and thanks for checking. TIL